### PR TITLE
Removed direct LAPACK calls

### PR DESCRIPTION
**Pull request by thomas-niehaus**
Echanged direct calls to LAPACK by the provided wrappers.

### CODE REVIEW

**Comment by thomas-niehaus** [comment]
Final version.

**Comment by thomas-niehaus** [comment]
Compiled and tested on dpdevel with gcc without errors. How can I get the config and compiler settings of the failing builds?

File: src/dftbp/timedep/linrespcommon.F90 [hunk 1]
```diff
@@ -2862,37 +2863,27 @@ subroutine calcMatrixSqrt(matIn, spaceDim, memDim, workArray, workDim, matOut, m
...
+    call heev(dummyM, dummyEV, 'U', 'V') 
 
     ! Calc. sqrt
     do ii = 1, spaceDim
       dummyM2(:,ii) = sqrt(dummyEV(ii)) * dummyM(:,ii)
     end do
 
-    call dgemm('N', 'T', spaceDim, spaceDim, spaceDim, 1.0_dp, dummyM2, spaceDim, dummyM,&
-        & spaceDim, 0.0_dp, matOut, memDim)
+    call gemm(matOut(:spaceDim,:spaceDim), dummyM2, dummyM, transB='T')
```
Comment (bhourahine) [inline src/dftbp/timedep/linrespcommon.F90:2879]: Probably better to extend the gemm interface to include setting LDA for the matrix, as the change as is creates a temporary copy.

**Comment by bhourahine** [comment]
@thomas-niehaus Look in /home/buildbot/testers/dftbplus/environments/ for specific instances with env.sh and arch.cmake files.

**Comment by thomas-niehaus** [comment]
Hi Ben, on dpdevel I do not see any predefined gfortran environements. The  (ubuntu-latest, nompi, Debug, 13) is for example not available.

File: src/dftbp/timedep/linrespcommon.F90 (hunk 1 above)
Comment (thomas-niehaus) [reply src/dftbp/timedep/linrespcommon.F90:2876]: You are right. I completely redesigned the weird memory setup in Stratmann. Now temp arrays are avoided and arrays are allocated to their required size. This simplifies a lot of gemm calls. Code is 100 % faster in the limited cases I studied and does not need more memory.

**Comment by thomas-niehaus** [comment]
I modified heev/dsyev to leave exit to the calling routine with an optional info parameter. For TD-DFTB info contains important information. If this is agreed on, I will also modify the other ssyev... routines.

**Comment by bhourahine** [comment]
@thomas-niehaus What order of merging is this with respect to [#1515](https://github.com/dftbplus/dftbplus/pull/1515)?  This PR already partially adds info, so is there any conflicts?

**Comment by thomas-niehaus** [comment]
The order is chronological. This PR is based on main, #1507 is based on this one, #1511 is based on #1507. To minimize conflicts they should be merged in this order. This one first.

**Comment by aradi** [comment]
Supseeded by #1522

**Pull request closed without merging**